### PR TITLE
Fixed #127

### LINKED ISSUE #127: Add unit-test coverage for PolicyManager object
We don't have any unit-test coverage for PolicyManager class.

Here's what would be expected to resolve this issue...

1.) Create a file spec/ssh_scan/policy_manager_spec.rb
2.) Add various tests that exercise examples of policy compliance and non-compliance to ensure it works as expected.  (this should also include error cases, like what happens when we're missing the information we need)

### PR DESCRIPTION
**Pull request by flash1452**
@claudijd This is my first PR, so please review it and suggest me any improvements in coding style.
Also, suggest me some other unit tests that I can include.

### CODE REVIEW

File: spec/ssh_scan/policy_manager_spec.rb [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	context "when checking out of policy encryption" do
+		yaml_string = "---\nname: Mozilla Intermediate\nkex:\n" +
+                  "- diffie-hellman-group-exchange-sha256\n" +
+                  "encryption:\n- aes256-ctr\n- aes192-ctr\n" +
+                  "- aes128-ctr\nmacs:\n- hmac-sha2-512\n" +
+                  "- hmac-sha2-256\ncompression:\n- none\n" +
+                  "- zlib@openssh.com\n" +
+                  "references:\n- https://wiki.mozilla.org/Security/Guidelines/OpenSSH\n"
+    result =  {
+			:encryption_algorithms_client_to_server => [
```
Comment (jinankjain) [inline spec/ssh_scan/policy_manager_spec.rb]: @flash145 Indentation Error

File: spec/ssh_scan/policy_manager_spec.rb [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+		    "chacha20-poly1305@openssh.com",
+		    "aes128-ctr",
+		    "aes192-ctr",
+		    "aes256-ctr",
+		    "aes128-gcm@openssh.com",
+		    "aes256-gcm@openssh.com"
+			],
+			:encryption_algorithms_server_to_client => [
```
Comment (jinankjain) [inline spec/ssh_scan/policy_manager_spec.rb]: @flash145 Indentation Error

**Comment by jinankjain** [comment]
@flash1452 There is a general indentation error at some place could you please go through them and send an updated PR

File: spec/ssh_scan/policy_manager_spec.rb [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      missing_policy_mac = policy_obj.missing_policy_macs()
+    end
+  end
+
+end
+
+
+
+
+
```
Comment (jinankjain) [inline spec/ssh_scan/policy_manager_spec.rb]: remove this extra spaces @flash1452

**Comment by jinankjain** [comment]
@flash1452 You have added the .swp file in this new patch could you please remove it.

**Comment by claudijd** [comment]
agreed, nuke the .swp file and this is landable.

**Comment by jinankjain** [comment]
@flash1452 One more thing could you please combine all the commit into one commit

**Comment by claudijd** [comment]
I may have spoke to soon, I'll be adding some comments on the code.

File: spec/ssh_scan/policy_manager_spec.rb [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                  "aes256-ctr",
+                  "aes128-gcm@openssh.com",
+                  "aes256-gcm@openssh.com"
+                ]
+              }
+
+    it "should load all the attributes properly" do
+      policy = SSHScan::Policy.from_string(yaml_string)
+      policy_obj = SSHScan::PolicyManager.new(result,policy)
+      missing_policy_enc = policy_obj.missing_policy_encryption()
```
Comment (claudijd) [inline spec/ssh_scan/policy_manager_spec.rb]: I would expect to see "expect" blocks here that actually assert what we're expecting to get back..

Example

    expect(policy_obj.missing_policy_encryption).to eql(...........STUFF.....)

File: spec/ssh_scan/policy_manager_spec.rb [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                  "aes192-ctr",
+                  "aes256-ctr",
+                  "aes128-gcm@openssh.com",
+                  "aes256-gcm@openssh.com"
+                ]
+              }
+
+    it "should load all the attributes properly" do
+      policy = SSHScan::Policy.from_string(yaml_string)
+      policy_obj = SSHScan::PolicyManager.new(result,policy)
```
Comment (claudijd) [inline spec/ssh_scan/policy_manager_spec.rb]: In ruby, all things are objects, so naming something with "_obj" is redundant.  I would just call it "policy_manager".

File: spec/ssh_scan/policy_manager_spec.rb [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                  "hmac-sha2-256",
+                  "hmac-sha2-512",
+                  "hmac-sha1"
+                ]
+              }
+
+    it "should load all the attributes properly" do
+      policy = SSHScan::Policy.from_string(yaml_string)
+      policy_obj = SSHScan::PolicyManager.new(result,policy)
+      out_of_policy_mac = policy_obj.out_of_policy_macs()
```
Comment (claudijd) [inline spec/ssh_scan/policy_manager_spec.rb]: needs expect

File: spec/ssh_scan/policy_manager_spec.rb [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+                  "aes256-ctr",
+                  "aes128-gcm@openssh.com",
+                  "aes256-gcm@openssh.com"
+                ]
+              }
+
+    it "should load all the attributes properly" do
+      policy = SSHScan::Policy.from_string(yaml_string)
+      policy_obj = SSHScan::PolicyManager.new(result,policy)
+      out_of_policy_enc = policy_obj.out_of_policy_encryption()
```
Comment (claudijd) [inline spec/ssh_scan/policy_manager_spec.rb]: needs expect

File: spec/ssh_scan/policy_manager_spec.rb [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+                  "hmac-sha2-256",
+                  "hmac-sha2-512",
+                  "hmac-sha1"
+                ]
+              }
+
+    it "should load all the attributes properly" do
+      policy = SSHScan::Policy.from_string(yaml_string)
+      policy_obj = SSHScan::PolicyManager.new(result,policy)
+      missing_policy_mac = policy_obj.missing_policy_macs()
```
Comment (claudijd) [inline spec/ssh_scan/policy_manager_spec.rb]: needs expect

**Comment by claudijd** [comment]
Ok, I'm done with review.

File: spec/ssh_scan/policy_manager_spec.rb (hunk 4 above)
Comment (claudijd) [inline spec/ssh_scan/policy_manager_spec.rb]: currently, all we're verifying is that it won't throw an exception.

File: spec/ssh_scan/policy_manager_spec.rb (hunk 8 above)
Comment (flash1452) [inline spec/ssh_scan/policy_manager_spec.rb]: @claudijd Should I remove the "missing_policy_mac = policy_obj.missing_policy_macs() / encryption()" lines and instead use expect assertions???

Comment (claudijd) [inline spec/ssh_scan/policy_manager_spec.rb]: Yes, please

**Comment by flash1452** [comment]
@claudijd @jinankjain Please review the assert statements.

**Comment by claudijd** [comment]
@flash1452 what happened, I don't see a code diff anymore?

**Comment by claudijd** [comment]
nvmd, I see you opened another PR.  Thanks!

**Comment by jinankjain** [comment]
@claudijd I merged the new PR which he sent with the suggested changes.

**Comment by claudijd** [comment]
thanks @jinankjain

**Pull request closed without merging**